### PR TITLE
Add benchmark tests to node datastore

### PR DESCRIPTION
**Pull request by md2119**
## Description

Package refactor (first commit) and benchmark tests (second commit).
 
Benchmark results:
```
After:
BenchmarkNodes/upsertNodeWithOldScan
BenchmarkNodes/upsertNodeWithOldScan-8         	       9	 112228287 ns/op
BenchmarkNodes/upsertNodeWithNewScan
BenchmarkNodes/upsertNodeWithNewScan-8         	       8	 312697667 ns/op
BenchmarkNodes/searchAll
BenchmarkNodes/searchAll-8                     	1000000000	         0.0001743 ns/op
BenchmarkNodes/searchForCluster
BenchmarkNodes/searchForCluster-8              	1000000000	         0.0006685 ns/op
BenchmarkNodes/deleteForClusters
BenchmarkNodes/deleteForClusters-8             	   13730	     80150 ns/op

Before:
BenchmarkNodes/upsertNodeWithOldScan
BenchmarkNodes/upsertNodeWithOldScan-8         	       8	 160853915 ns/op
BenchmarkNodes/upsertNodeWithNewScan
BenchmarkNodes/upsertNodeWithNewScan-8         	       7	 166447309 ns/op
BenchmarkNodes/searchAllStoresGetStoresAndListNodes
BenchmarkNodes/searchAllStoresGetStoresAndListNodes-8         	1000000000	         0.0001730 ns/op
BenchmarkNodes/searchForClusterWithListNodes
BenchmarkNodes/searchForClusterWithListNodes-8                	1000000000	         0.0007372 ns/op
BenchmarkNodes/deleteForClustersWithRemoveClusterNodeStore
BenchmarkNodes/deleteForClusters-8             	    7135	    223740 ns/op

```

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed
CI

### CODE REVIEW

File: central/node/datastore/datastore_impl.go [hunk 1]
```diff
@@ -235,7 +235,8 @@ func (ds *datastoreImpl) deleteNodeFromStore(ctx context.Context, ids ...string)
 			continue
 		}
 		if err := ds.risks.RemoveRisk(deleteRiskCtx, id, storage.RiskSubjectType_NODE); err != nil {
-			return err
+			errorList.AddError(err)
+			continue
 		}
 	}
 	// removing component risk handled by pruning
```
Comment (RTann) [inline central/node/datastore/datastore_impl.go:242]: not you, but this comment doesn't make sense to me down here

File: central/node/datastore/datastore_impl.go [hunk 2]
```diff
@@ -235,7 +235,8 @@ func (ds *datastoreImpl) deleteNodeFromStore(ctx context.Context, ids ...string)
 			continue
 		}
 		if err := ds.risks.RemoveRisk(deleteRiskCtx, id, storage.RiskSubjectType_NODE); err != nil {
-			return err
+			errorList.AddError(err)
```
Comment (RTann) [inline central/node/datastore/datastore_impl.go:238]: interesting, I wonder if this was on purpose or just an accident?

**Review by RTann — APPROVED** [review]
Forgot to say it in the last message, but just change the one nit since `datastore` is too generic for that file, then :shipit:

File: central/cluster/datastore/datastore_test_constructors.go [hunk 3]
```diff
@@ -18,8 +18,7 @@ import (
 	networkBaselineManager "github.com/stackrox/rox/central/networkbaseline/manager"
 	netEntityDataStore "github.com/stackrox/rox/central/networkgraph/entity/datastore"
 	netFlowsDataStore "github.com/stackrox/rox/central/networkgraph/flow/datastore"
-	nodeDataStore "github.com/stackrox/rox/central/node/datastore/dackbox/datastore"
-	nodeNodeDataStore "github.com/stackrox/rox/central/node/datastore/dackbox/datastore"
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/cluster/datastore/datastore_test_constructors.go:22]: 🙂  Yes, seems like aliasing hides it. This was captured by Goland refactor functionality so I noticed it.

File: central/graphql/resolvers/resolver.go [hunk 4]
```diff
@@ -37,7 +37,7 @@ import (
 	namespaceDataStore "github.com/stackrox/rox/central/namespace/datastore"
 	nfDS "github.com/stackrox/rox/central/networkgraph/flow/datastore"
 	npDS "github.com/stackrox/rox/central/networkpolicies/datastore"
-	nodeDataStore "github.com/stackrox/rox/central/node/datastore/dackbox/datastore"
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/graphql/resolvers/resolver.go:40]: Will do. This was adjusted by Goland's refactor functionality.

File: central/node/datastore/datastore_impl.go (hunk 2 above)
Comment (md2119) [reply central/node/datastore/datastore_impl.go:238]: I believe accident. Furthermore, the pattern is not consistent between two write ops within same block.

**Comment by md2119** [comment]
@RTann updated the benchmark results. The node pruning for the cluster improved dramatically. Post cluster remove op was starting point for the refactor effort.

**Pull request merged**